dashboard/test/controllers/authored_hint_view_requests_controller_test.rb

```diff
@@ -72,4 +72,43 @@ class AuthoredHintViewRequestsControllerTest < ActionController::TestCase
       end
     end
   end
+
+  test 'records all (optional) data fields' do
+    data = {
+      scriptId: @script.id,
+      levelId: @level.id,
+      hintId: "first",
+      hintClass: 'bottom-out',
+      hintType: 'general',
+      prevTime: 1,
+      prevAttempt: 2,
+      prevTestResult: 3,
+      prevActivityId: 4,
+      prevLevelSourceId: 5,
+      nextTime: 6,
+      nextAttempt: 7,
+      nextTestResult: 8,
+      nextActivityId: 9,
+      nextLevelSourceId: 10,
+      finalTime: 11,
+      finalAttempt: 12,
+      finalTestResult: 13,
+      finalActivityId: 14,
+      finalLevelSourceId: 15
+    }
+
+    post(
+      :create,
+      params: {
+        hints: [data]
+      },
```

Comment:
super nit: At a minimum, I'd suggest keeping all of `params` on one line (removing the line breaks). If it fits, the whole `post`.